packages/unified-latex-ctan/utils/argument-globber.ts

```diff
@@ -0,0 +1,72 @@
+import { arg } from "@unified-latex/unified-latex-builder";
+import { Argument, Node } from "@unified-latex/unified-latex-types";
+import { parse as parseArgspec } from "@unified-latex/unified-latex-util-argspec";
+import { gobbleSingleArgument } from "@unified-latex/unified-latex-util-arguments";
+import { match } from "@unified-latex/unified-latex-util-match";
+
+export function getOptionalArg(
+    nodes: Node[],
+    pos: number
+): { arg: Argument; nodesRemoved: number } {
+    const { argument, nodesRemoved } = gobbleSingleArgument(
+        nodes,
+        parseArgspec("o")[0],
+        pos
+    );
+    return {
+        arg: argument || arg([], { openMark: "", closeMark: "" }),
+        nodesRemoved,
+    };
+}
+
+export function getGroup(
+    nodes: Node[],
+    pos: number
+): { arg: Argument; nodesRemoved: number } {
+    const nextArgNode = nodes[pos];
+    if (match.group(nextArgNode)) {
+        nodes.splice(pos, 1);
+
+        return {
+            arg: arg(nextArgNode.content),
+            nodesRemoved: 1,
+        };
+    }
+
+    return {
+        arg: arg([], { openMark: "", closeMark: "" }),
+        nodesRemoved: 0,
+    };
+}
+
+export function getDelimGroup(
+    nodes: Node[],
+    pos: number
+): { arg: Argument; nodesRemoved: number } {
```

Comment:
I think this is already accomplished with the argspec `d##`. Since the delimiter can change, you might need to do something like `d${delim}${delim}`

Reply:
I'm not aware of the use `d${delim}${delim}`. Will check on that.

Reply:
Also, neither `d` nor `r` or `v` was already handled in `gobbleSingleArgument`, or I missed something.

Reply:
Looks like you're right! Let's go ahead and add `d` support to `gobbleSingleArgument`. Basically, if it can be supported there, we should add it there to make it easier for the next people trying to support a fancy macro :-)

Reply:
I do missed something! Figured out.

Reply:
> Looks like you're right! Let's go ahead and add `d` support to `gobbleSingleArgument`. Basically, if it can be supported there, we should add it there to make it easier for the next people trying to support a fancy macro :-)

Indeed, `r` and `d` were so well supported in `gobbleSingleArgument`! That's really amazing!